internal/cli/config/config.go

```diff
@@ -204,3 +247,27 @@ func Builder() *cobra.Command {
 
 	return cmd
 }
+
+// atlasProjects transform []*atlas.Project to a map[string]string and []string
+func atlasProjects(projects []*atlas.Project) (pMap map[string]string, pSlice []string) {
+	pMap = make(map[string]string, len(projects))
+	pSlice = make([]string, len(projects))
+	for i, p := range projects {
+		d := fmt.Sprintf(nameIDFormat, p.Name, p.ID)
+		pMap[d] = p.ID
+		pSlice[i] = d
+	}
+	return pMap, pSlice
+}
+
+// omProjects transform []*opsmngr.Project to a map[string]string and []string
+func omProjects(projects []*opsmngr.Project) (pMap map[string]string, pSlice []string) {
+	pMap = make(map[string]string, len(projects))
+	pSlice = make([]string, len(projects))
+	for i, p := range projects {
+		d := fmt.Sprintf(nameIDFormat, p.Name, p.ID)
+		pMap[d] = p.ID
+		pSlice[i] = d
+	}
+	return pMap, pSlice
+}
```

Comment:
[q] Is it possible to make a common helper here? The code looks almost identical except for `*atlas.Project` vs `*opsmngr.Project` in the signature.

Reply:
sadly I don't know how, golang does not have great support for generic slices, so the for would need to be duplicated given that I would not be able to do `p.ID` for two different types even when they gave the same properties (supposedly [they are working on this](https://blog.golang.org/why-generics))

`pMap = make(map[string]string, len(projects))` and such are also troublesome  since I can't do len of the interface unless I cast it, which just makes me extract this from this method but duplicated somewhere else 

so is just a question of where do we duplicate :(

Reply:
Fair enough, no language is perfect. Thanks for the response for my own learning.